geth/notification/payload.go

```diff
@@ -1,12 +1,40 @@
 package notification
 
+import "github.com/NaySoftware/go-fcm"
+
 // Payload data of message.
 type Payload struct {
-	Title string
-	Body  string
-	Icon  string
-	Sound string
-	Badge string
-	Tag   string
-	Color string
+	Title string `json:"title,omitempty"`
+	Body  string `json:"body,omitempty"`
+	Icon  string `json:"icon,omitempty"`
+	Sound string `json:"sound,omitempty"`
+	Badge string `json:"badge,omitempty"`
+	Tag   string `json:"tag,omitempty"`
+	Color string `json:"color,omitempty"`
+	// this are leftovers from original fcm.NotificationPayload
+	ClickAction      string `json:"click_action,omitempty"`
+	BodyLocKey       string `json:"body_loc_key,omitempty"`
+	BodyLocArgs      string `json:"body_loc_args,omitempty"`
+	TitleLocKey      string `json:"title_loc_key,omitempty"`
+	TitleLocArgs     string `json:"title_loc_args,omitempty"`
+	AndroidChannelID string `json:"android_channel_id,omitempty"`
+}
+
+// ToFCMNotificationPayload turns Payload into fcm.NotificationPayload
+func (p *Payload) ToFCMNotificationPayload() *fcm.NotificationPayload {
```

Comment:
Maybe FirebaseClient (1) should know how to handle `notification.Payload` (2) instead of making (2) aware of (1)?